src/main/java/org/example/model/User.java

```diff
@@ -0,0 +1,92 @@
+package org.example.model;
+
+import org.apache.log4j.Logger;
+import org.example.model.Observer.DesiredPath;
+
+import java.util.Objects;
+
+public class User {
+    private static final Logger logger = Logger.getLogger("GLOBAL");
+    private Integer id;
+    private String name;
+    private String surname;
+    private String email;
+    private DesiredPath desiredPath;
+
+    public User() {
+    }
+
+    public User(Integer id, String name, String surname, String email) {
+        this.id = id;
+        this.name = name;
+        this.surname = surname;
+        this.email = email;
+    }
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getSurname() {
+        return surname;
+    }
+
+    public void setSurname(String surname) {
+        this.surname = surname;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        User user = (User) o;
+        return Objects.equals(id, user.id) && Objects.equals(name, user.name) && Objects.equals(surname, user.surname) && Objects.equals(email, user.email);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, name, surname, email);
+    }
+
+    @Override
+    public String toString() {
+        return "User{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                ", surname='" + surname + '\'' +
+                ", email='" + email + '\'' +
+                '}';
+    }
+
+    public DesiredPath getDesiredPath() {
+        return desiredPath;
+    }
+
+    public void setDesiredPath(DesiredPath desiredPath) {
+        this.desiredPath = desiredPath;
+    }
+
+    public void update() {
+        logger.info(String.format("Your destination is %s m away", getDesiredPath().getRoute().getDistance()));
```

Comment:
For now, this is good but logically speaking, this will always return the same distance since the route's start point and end point are always the same. Will need to figure out how get the distance based on the DesiredPath's current location and the route's end point (or DesiredPath's destination).

Reply:
Yeah, but the route stored in desired path changes. We could change that to an array of routes and get the combined distance later as well, depending on our business logic, but my goal here was to have the Route (or what have you) stored in DesiredPath changed based off user preferences.

Reply:
Ohh I see. I assumed that it would be based off of Desired Path because its currentLocation attribute would track where the User is but your logic makes sense.